<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:macro name="fri_gpio_config">
    <gpio name="gpio">
      <command_interface name="Input1" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input2" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input3" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input4" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input5" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input6" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input7" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input8" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input9" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>
      <command_interface name="Input10" data_type="boolean">
         <param name="initial_value">0</param>
      </command_interface>

      <state_interface name="Output1" data_type="boolean"/>
      <state_interface name="Output2" data_type="boolean"/>
      <state_interface name="Output3" data_type="boolean"/>
      <state_interface name="Output4" data_type="boolean"/>
      <state_interface name="Output5" data_type="boolean"/>
      <state_interface name="Output6" data_type="boolean"/>
      <state_interface name="Output7" data_type="boolean"/>
      <state_interface name="Output8" data_type="boolean"/>
      <state_interface name="Output9" data_type="boolean"/>
      <state_interface name="Output10" data_type="boolean"/>
    </gpio>
  </xacro:macro>
</robot>
